File: src/main/java/org/willxu/algorithm/service/integer/LengthOfLastWord.java
/*
 * Copyright (C) 2021  Weiyang(Will) Xu
 * 
 * This file is part of algorithm.
 * Algorithm is free software: you can redistribute it and/or modify
 * it under the term of the GNU General Public License version 3 or
 * any later version, as specified in the readme.md file.
 */

package org.willxu.algorithm.service.integer;

public interface LengthOfLastWord {
    /**
     * Given a string s consists of some words separated by spaces,
     * return the length of the last word in the string. If the last
     * word does not exist, return 0.
     *
     * A word is a maximal substring consisting of non-space characters
     * only.
     *
     * @param s 1 <= s.length <= 10^4
     *          s consists of only English letters and spaces ' '.
     * @return the length of last word
     */
    int lengthOfLastWord(String s);
}

File: src/leetcode/p0000_0099/p0058_length_of_last_word/length_of_last_word.py
from abc import ABC, abstractmethod

class LengthOfLastWord(ABC):
    """
    Given a string s consisting of words and spaces, return the length of the last word in the string.

    A word is a maximal substring consisting of non-space characters only.
    """
    @abstractmethod
    def lengthOfLastWord(self, s: str) -> int:
        """
        Args:
            s:
                1 <= s.length <= 10^4
                s consists of only English letters and spaces ' '.
                There will be at least one word in s.
        """
        pass
